TripalCultivate/TripalCultivate-Phenotypes
# g4.91 Bring up to standards code pertaining to Validator upgrade

**PART 2 of Issue #91**

## Motivation
This PR aims to better meet Drupal/PHP Standards, which will serve as a standard moving forward in our development for all TripalCultivate modules.

We'll do this by enabling the PHP Sniffer & Beautify extension for VS code as a guide. This includes removing unnecessary use statements, using proper if/elseif statements, proper dependency injection, etc. NOTE: We may not agree with ALL standard suggestions and/or some may be way too tedious to implement for little gain. I will document in the next section what standards specifically are being enforced that weren't previously.

## What does this PR do?
1. Adds in checking for standards to the devcontainer including using the TripalCultivate-specific standards that can be accessed here: https://github.com/TripalCultivate/TripalCultivate/blob/4.x/.codingstandards.xml
2. Applies PHP/Drupal standards to all development pertaining to VALIDATORS. Specifically, **_everything_** (including sub-directories) in the following directories:
   - `trpcultivate_phenotypes/src/Plugin/Validators`
   - `trpcultivate_phenotypes/src/TripalCultivateValidator`
   - `trpcultivate_phenotypes/tests/src/Kernel/Validators`

   Some of the most common changes made to a majority of the files were:
   - Fixes to improper indenting in comments
   - Commas added to the end of arrays
   - `elseif` and `catch` in a try-catch need to start on a new line
   - Removed spacing between variables and open brackets (ex. $array[ $key ])
   - Empty line added between the final method and the closing bracket for a class
   - Re-ordering of use statements (a couple of unnecessary statements removed)
   - Data types in @var tags need to be fully namespaced
   - Proper spacing after if and foreach statements
   - And more! 🤪 

3. Additional updates to docblocks were made to include @param/@return/@throws tags where applicable, with attempts to make the text more succinct and improve clarity (including single-line comments). I particularly tried to standardize the documentation for each of the validators above their validate methods and for the methods in traits. This is where I think focussing review would be most valuable 😄 

## Testing

### Automated Testing
No automated tests have been added, though it's good to confirm that every test is still passing.

### Manual Testing
To verify that VS code is automatically enforcing the new standards:
1. Create a new docker from this branch and open the repo in vs code.
2. Go to `Extensions` in the left sidebar (the icon with building blocks) and check that `PHP Sniffer & Beautifier` shows up in either your `Local - Installed` or `Dev Container` lists.
3. Choose any one of the files changed in this PR, and confirm that no problems are reported in your terminal window
4. To compare with a file not yet standardized, select one that is not in the scope of this PR (I suggest `/trpcultivate_phenotypes/src/Plugin/TripalImporter/TripalCultivatePhenotypesTraitsImporter.php`). You should see problems reported in the terminal window section of VS code. 
5. Then, go to the very top bar and select `View > Command Palette` and a search box should pop up. Start typing "Format" and select `Format Document`. This will automatically apply changes from `PHP Sniffer & Beautifier` and significantly lower the number of problems and warnings. (Don't commit these changes yet - that is for a future PR! 😅)

After confirming that the `PHP Sniffer & Beautifier` is working as expected, visually inspect the changes that occurred in each file as part of this PR. Considering the multitude of total files changed, perhaps it is more feasible to randomly select one file from each subdirectory since all files in a particular directory have very similar changes.

I _highly_ recommend you specifically check any methods you worked on in the past and take a look at changes I made to docblocks and comments. Please feel free to make suggestions if you don't agree with my changes 😄 🙏
